### ISSUE TITLE
osx serial ports do not show port id

### ISSUE THREAD

**Issue by tablatronix** [gepd/Deviot#214]
I cannot tell which device is which, I love seeing the names, but when there are dups, its a problem

![screen shot 2018-03-08 at 10 10 22 pm](https://user-images.githubusercontent.com/807787/37190127-ac06f960-231d-11e8-9744-701a71dbbe2a.png)

**Comment by gepd** [comment]
I'll try to do this tomorrow I don't have a OSX or Linux machine to change this

**Comment by tablatronix** [comment]
Let me know where to start. I need to learn python at some point

**Comment by gepd** [comment]
It was easier than I thought, It should be resolved with https://github.com/gepd/Deviot/commit/3f92c9d0dc88d262f98027896bc4d07ed6730c5a, let me know if that is what you expect

**Comment by tablatronix** [comment]
This seems to be a breaking change, tested on windows, it says select port over and over in a loop.

**Comment by gepd** [comment]
Fixed

**Comment by tablatronix** [comment]
This seems to work and help, although its still a bit of a mystery lol
Can probably close.

![screen shot 2018-06-13 at 7 05 09 pm](https://user-images.githubusercontent.com/807787/41384713-0d6d5d82-6f3d-11e8-9270-c9c092dae636.png)

**Comment by gepd** [comment]
I see, I think it would be always confusing when you have multiples devices with the same spec.. 
A solution would be to add an alias, I could look for a kind of ID that will make possible to always assign an alias that can replace or be merged with the current name

**Comment by tablatronix** [comment]
Yeah this is on my list to look into, I will look into having some kind of tty alias or have a naming convention, Not sure where OSX is getting the 15,16,17 suffixes but it must be caching them somewhere.

It is mostly an edge case.

**Comment by gepd** [comment]
What about having an option that when it's active, it asks for an alias everytime you connect the device, I don't see other way to do it

**Comment by tablatronix** [comment]
How would that work? How about indicating the last used port, colored highlighted? Might help a little bit. 

I have another issue where some usb shows 2 ports, ill check if its a pio issue or not

**Comment by gepd** [comment]
Yes it may be tedious

Idicanting the last used port where?

**Comment by tablatronix** [comment]
In the devices list, can it appear highlighted?

**Comment by gepd** [comment]
Oh now I understand, ST does not provide anyway to highlight it, but what I can do is to add a character at the beginning of the active ports.

```js
* CP2102 USB to UART Bridge Controller...
CP2102 USB to UART Bridge Controller...
* CP2102 USB to UART Bridge Controller...
```

In the previous case the first one and the last one are active.

What do you think?

**Comment by tablatronix** [comment]
yeah that might be cool, my main issue is that I have over 5 uarts, some are connected to test equipment like lab power supplies and multimeter, and some arduino etc, and of course they are all just slab_uart, or ftdi, etc, and it is hard to keep track of them lol

Even better would be remember the last device that was used and indicate it , probably requires saving stuff somewhere so might not be a good idea and prohibitive

**Comment by gepd** [comment]
To remember what was the last device used isn't a problem, but where should it be displayed?

And what about the cases where you have devices with the same name, until now I didn't find any unique parameter to identify each device
